crates/mun_hir/src/item_scope.rs

```diff
@@ -0,0 +1,87 @@
+use crate::builtin_type::BuiltinType;
+use crate::{ids::ItemDefinitionId, visibility::Visibility, Name, PerNs};
+use once_cell::sync::Lazy;
+use rustc_hash::FxHashMap;
+
+/// Holds all items that are visible from an item as well as by which name and under which
+/// visibility they are accessible.
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct ItemScope {
+    /// All types visible in this scope
+    types: FxHashMap<Name, (ItemDefinitionId, Visibility)>,
+
+    /// All values visible in this scope
+    values: FxHashMap<Name, (ItemDefinitionId, Visibility)>,
+
+    /// All items that are defined in this scope
+    defs: Vec<ItemDefinitionId>,
+}
+
+pub(crate) static BUILTIN_SCOPE: Lazy<FxHashMap<Name, PerNs<(ItemDefinitionId, Visibility)>>> =
```

Comment:
As we're renaming, should this be `PRIMITIVE` and `PrimitiveType::ALL`?

Reply:
I did leave the `BUILTIN_SCOPE` (because thats what it actually is), but I did rename `BuiltinType` to `PrimitiveType`